### PR TITLE
FLUX dev and lite models support

### PR DESCRIPTION
**Pull request by likholat**
```python
prompt = "A cat holding a sign that says hello world"
image_tensor = pipe.generate(
        prompt,
        num_inference_steps=5,
        generator=Generator(42) 
    )
```

Freepik/flux.1-lite-8B-alpha:

image

black-forest-labs/FLUX.1-dev:

image

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/cpp/src/image_generation/flux_pipeline.hpp [hunk 1]
```diff
@@ -269,6 +269,14 @@ class FluxPipeline : public DiffusionPipeline {
 
         ov::Tensor latent_image_ids = prepare_latent_image_ids(generation_config.num_images_per_prompt, height / 2, width / 2);
 
+        // add guidance tensor if input exist
+        auto input_names = m_transformer->get_config().m_model_input_names;
+        if (std::find(input_names.begin(), input_names.end(), "guidance") != input_names.end()) {
+            ov::Tensor guidance = ov::Tensor(ov::element::f32, {generation_config.num_images_per_prompt});
+            std::fill_n(guidance.data<float>(), guidance.get_size(), static_cast<float>(m_generation_config.guidance_scale));
```
Comment (ilya-lavrenov) [inline src/cpp/src/image_generation/flux_pipeline.hpp:276]:
```suggestion
            std::fill_n(guidance.data<float>(), guidance.get_size(), static_cast<float>(m_custom_generation_config.guidance_scale));
```

File: src/cpp/src/image_generation/models/flux_transformer_2d_model.cpp [hunk 2]
```diff
@@ -106,6 +117,7 @@ FluxTransformer2DModel& FluxTransformer2DModel::reshape(int batch_size,
 FluxTransformer2DModel& FluxTransformer2DModel::compile(const std::string& device, const ov::AnyMap& properties) {
     OPENVINO_ASSERT(m_model, "Model has been already compiled. Cannot re-compile already compiled model");
     ov::CompiledModel compiled_model = utils::singleton_core().compile_model(m_model, device, properties);
+    get_input_names(m_config.m_model_input_names, compiled_model.inputs());
```
[the remark this answers is not in the document]
Comment (likholat) [reply src/cpp/src/image_generation/models/flux_transformer_2d_model.cpp:120]: Have you tested `black-forest-labs/FLUX.1-schnell` model as well?

[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/cpp/src/image_generation/models/flux_transformer_2d_model.cpp:120]: yes

File: src/cpp/src/image_generation/flux_pipeline.hpp [hunk 3]
```diff
@@ -269,10 +269,14 @@ class FluxPipeline : public DiffusionPipeline {
...
         ov::Tensor latent_image_ids = prepare_latent_image_ids(generation_config.num_images_per_prompt, height / 2, width / 2);
 
+        ov::Tensor guidance = ov::Tensor(ov::element::f32, {generation_config.num_images_per_prompt});
+        std::fill_n(guidance.data<float>(), guidance.get_size(), static_cast<float>(generation_config.guidance_scale));
+
         m_transformer->set_hidden_states("pooled_projections", pooled_prompt_embeds);
         m_transformer->set_hidden_states("encoder_hidden_states", prompt_embeds);
         m_transformer->set_hidden_states("txt_ids", text_ids);
         m_transformer->set_hidden_states("img_ids", latent_image_ids);
+        m_transformer->set_hidden_states("guidance", guidance);
```
Comment (ilya-lavrenov) [inline src/cpp/src/image_generation/flux_pipeline.hpp:279]: should we add it based on `guidance_embeds` flag ?

Comment (likholat) [reply src/cpp/src/image_generation/flux_pipeline.hpp]: fixed

**Pull request merged**